components/airtable/new-modified-or-deleted-records.js

```diff
@@ -0,0 +1,111 @@
+const airtable = require('https://github.com/PipedreamHQ/pipedream/components/airtable/airtable.app.js');
+const moment = require('moment');
+const axios = require('axios');
+const Bottleneck = require('bottleneck');
+
+const limiter = new Bottleneck({
+  minTime: 200, // 5 requets per second
+});
+const axiosRateLimiter = limiter.wrap(axios);
+
+module.exports = {
+  name: 'New, Modified or Deleted Records',
```

Comment:
We like to add a `description` to event sources, as well (we need to add these to the other Airtable event sources!), something like "Emits an event each time a record is added, updated, or deleted in an Airtable table"

Reply:
have added the description field like you suggested.

Reply:
When we talked about this source as a team, two questions re: limits came up:

1) The `this.db` limits we discussed below
2) Listing all records for a table, like you're doing below, is the best way we know of to check for deleted records. That approach is fine, but for large tables, that may also yield issues (for example, making so many HTTP requests to list records at each offset may cause the component to timeout - currently that timeout is 30 seconds).

Overall, we think it's fine to ship this source as-is, but note an initial record limit for this source in the description so that users with large tables understand that they may hit issues.

I'd change the description to read:

"Emits an event each time a record is added, updated, or deleted in an Airtable table. Supports tables up to 10,000 records".

10,000 is just an estimate, but we can start there and if we see any issues with the source, we can revisit.

Reply:
have updated the description as you suggested to let users know of the limits.